shark-ai/export_and_serve.py

```diff
@@ -0,0 +1,412 @@
+                raise
+
+        if self.shard and not os.path.isdir(self.artifacts_dir):
+            logger.warning(
+                f"Artifacts directory {str(self.artifacts_dir)} does not exist, creating..."
+            )
+            try:
+                Path.mkdir(self.artifacts_dir, parents=True)
+            except:
+                logger.critical("Failed to create artifacts directory")
+                raise
+
+        if self.export:
+            self.mlir_path = Path(os.path.join(self.export_dir, "model.mlir"))
+            self.model_config_path = Path(os.path.join(self.export_dir, "config.json"))
+
+            logger.info("TP1 export is enabled")
+            logger.info(f"TP1 MLIR will be saved to {self.mlir_path}")
+            logger.info(f"TP1 model config will be saved to {self.model_config_path}")
+
+        if self.shard:
+            self.mlir_path = Path(os.path.join(self.export_dir, "model.mlir"))
+            self.model_config_path = Path(
+                os.path.join(self.export_dir, "config_tp8.json")
+            )
+
+            logger.info("TP8 sharding and export is enabled")
+            logger.info(f"Artifacts directory set to {self.artifacts_dir}")
+            logger.info(f"TP8 MLIR will be saved to {self.mlir_path}")
+            logger.info(f"TP8 model config will be saved to {self.model_config_path}")
+
+        if self.compile:
+            logger.info("Compilation is enabled")
+            if self.ir_path is not None:
+                logger.info(f"IR for compilation located at {self.ir_path}")
+                self.mlir_path = self.ir_path
+
+        logger.info(f"Exported model will be saved to {self.export_dir}")
+
+    def export_tp1(self) -> None:
+        batch_size_string = ",".join([str(b) for b in self.export_batch_sizes])
+        logger.info(f"Exporting with batch sizes {batch_size_string}")
+
+        # TODO(vinayakdsci): Refactor this into a separate function.
+        export_subp = subprocess.run(
+            [
+                "python3",
+                "-m",
+                "sharktank.examples.export_paged_llm_v1",
+                f"--gguf-file={self.weight_loc}",
+                f"--output-mlir={str(self.mlir_path)}",
+                f"--output-config={self.model_config_path}",
+                f"--bs={batch_size_string}",
+            ],
+            capture_output=True,
+        )
+
+        # Exit if the command fails, no point in throwing an exception.
+        if export_subp.returncode != 0:
+            logger.error("Failed to export model in TP1 mode")
+            print(export_subp.stderr.decode(), file=sys.stderr)
+            exit(export_subp.returncode)
+
+        logger.info(
+            f"Successfully exported and saved model MLIR and config to {self.export_dir}"
+        )
+
+    def export_tp8(self) -> None:
+        shard_file = os.path.join(
+            str(self.artifacts_dir),
+            str(self.weight_loc).split("/")[-1][:-5] + "_tp8.irpa",
+        )
```

Comment:
use pathlib here. Join with `/` and replace the `[-1][:-5]` with some of the functions from https://docs.python.org/3/library/pathlib.html#corresponding-tools (`.stem()`, `.parent()`, etc.)